### ISSUE TITLE
[x64] Test failed: tracing_eventsource._eventsourcetrace_eventsourcetrace_eventsourcetrace_._eventsourcetrace_eventsourcetrace_eventsourcetrace_sh

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#23273]
**Job:**
https://ci.dot.net/job/dotnet_coreclr/job/master/view/x64/job/jitstress/job/x64_checked_osx10.12_jitstress2_tst/552/

**Failed tests:**
tracing_eventsource._eventsourcetrace_eventsourcetrace_eventsourcetrace_._eventsourcetrace_eventsourcetrace_eventsourcetrace_sh

**Log:**
```
tracing_eventsource._eventsourcetrace_eventsourcetrace_eventsourcetrace_._eventsourcetrace_eventsourcetrace_eventsourcetrace_sh [FAIL]
        
        Assert failure(PID 26419 [0x00006733], Thread: 559254 [0x88896]): Assertion failed 'kind == Overwrite' in 'Tracing.Tests.EventSourceTrace:Main(ref):int' (IL size 411)
        
            File: /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12/src/jit/jithashtable.h Line: 270
            Image: /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_jitstress2_tst/bin/tests/OSX.x64.Checked/Tests/Core_Root/corerun
        
        /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_jitstress2_tst/bin/tests/OSX.x64.Checked/tracing/eventsource/eventsourcetrace/eventsourcetrace/eventsourcetrace.sh: line 259: 26419 Abort trap: 6           (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"
        /n/nReturn code:      1/nRaw output file:      /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_jitstress2_tst/bin/tests/OSX.x64.Checked/tracing/eventsource/Reports/tracing.eventsource/eventsourcetrace/eventsourcetrace/eventsourcetrace.output.txt/nRaw output:/nBEGIN EXECUTION
        /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_jitstress2_tst/bin/tests/OSX.x64.Checked/Tests/Core_Root/corerun eventsourcetrace.exe ''
        Expected: 100
        Actual: 134
        END EXECUTION - FAILED
        Test Harness Exitcode is : 1
        /nTo run the test:/n> set CORE_ROOT=/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_jitstress2_tst/bin/tests/OSX.x64.Checked/Tests/Core_Root/n> /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_jitstress2_tst/bin/tests/OSX.x64.Checked/tracing/eventsource/eventsourcetrace/eventsourcetrace/eventsourcetrace.sh/n
        Expected: True
        Actual:   False
        Stack Trace:
          /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12/bin/tests/OSX.x64.Checked/TestWrappers/tracing.eventsource/tracing.eventsource.XUnitWrapper.cs(109,0): at tracing_eventsource._eventsourcetrace_eventsourcetrace_eventsourcetrace_._eventsourcetrace_eventsourcetrace_eventsourcetrace_sh()
```

**Comment by sandreenko** [comment]
cc @dotnet/jit-contrib 

I see this regression in desktop testing.

**Comment by sandreenko** [comment]
It was fixed by #23272.
